schemas/vulnerability.yml

```diff
@@ -0,0 +1,176 @@
+      level: extended
+      type: float
+      short: Vulnerability Base score.
+      description: >
+        Scores can range from 0.0 to 10.0, with 10.0 being the most severe.
+
+        Base scores cover an assessment for exploitability metrics (attack vector,
+        complexity, privileges, and user interaction), impact metrics (confidentialy,
+        integrity, and availability), and scope.
+        For example (https://www.first.org/cvss/specification-document)
+
+      example: 5.5
+
+    - name: score.temporal
+      level: extended
+      type: float
+      short: Vulnerability Temporal score.
+      description: >
+        Scores can range from 0.0 to 10.0, with 10.0 being the most severe.
+
+        Temporal scores cover an assessment for code maturity, remediation level, and
+        confidence.
+        For example (https://www.first.org/cvss/specification-document)
+
+    - name: score.environmental
+      level: extended
+      type: float
+      short: Vulnerability Environmental score.
+      description: >
+        Scores can range from 0.0 to 10.0, with 10.0 being the most severe.
+
+        Environmental scores cover an assessment for any modified Base metrics,
+        confidentiality, integrity, and availability requirements.
+        For example (https://www.first.org/cvss/specification-document)
+
+      example: 5.5
+
+    - name: score.version
+      level: extended
+      type: keyword
+      short: CVSS version.
+      description: >
+        The National Vulnerability Database (NVD) provides qualitative severity rankings
+        of "Low", "Medium", and "High" for CVSS v2.0 base score ranges in addition to the
+        severity ratings for CVSS v3.0 as they are defined in the CVSS v3.0 specification.
+
+        CVSS is owned and managed by FIRST.Org, Inc. (FIRST), a US-based non-profit
+        organization, whose mission is to help computer security incident response teams
+        across the world.
+        For example (https://nvd.nist.gov/vuln-metrics/cvss)
+
+      example: 2.0
+
+    - name: category
+      level: extended
+      type: keyword
+      short: Category of a vulnerabilty.
+      description: >
+        The type of system or architecture that the vulnerability affects. These may be
+        platform-specific (for example, Debian or SUSE) or general (for example, Database
+        or Firewall).
+        For example (https://qualysguard.qualys.com/qwebhelp/fo_portal/knowledgebase/vulnerability_categories.htm)
+
+      example: Firewall
+
+    - name: description
+      level: extended
+      type: keyword
+      short: Description of the vulnerability.
+      description: >
+        The description of the vulnerability that provides additional context of the
+        vulnerability.
+        For example (https://cve.mitre.org/about/faqs.html#cve_entry_descriptions_created)
+
+      example: In macOS before 2.12.6, there is a vulnerability in the RPC...
+
+    - name: scanner.id
```

Comment:
This field name bugs me. It sounds like an internal ID for a single finding, when actually the example shows the public ID of a vulnerability.

If this is meant to be the public ID, I'd have this straight at `vulnerability.id`

Reply:
Agreed. Makes sense.